app/controllers/users_controller.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def shopping_list
+    @foods = Food.all
+    @recipe_foods = Food.joins(:recipes).distinct
+    @shopping_list = []
+    @recipe_foods.each do |recipe_food|
+      @shopping_list << recipe_food
+    end
+    @final_list = @foods.where(id: @shopping_list.map(&:id))
+
+    @price = @final_list.pluck(:price)
+    @quantity = @final_list.pluck(:quantity)
+    @total = @price.zip(@quantity).map { |x, y| x * y }.sum
```

Comment:
- [ ] As requested by the previous reviewer, please fix the shopping list. Right now, the list shows the exact quantity of the item as an ingredient for a recipe. But the foods belong to a single user. Think of it as a stock. So I already have that ingredient as 500 grams. But recipe wants me to add 750gr. 

When the shopping list generated, user should see of the ingredient needed as 250 grams. The price should re-calculated. Kindly add this functionality.

Reply:
well resolved! ✔️